docs/source/contributing.rst

```diff
@@ -0,0 +1,187 @@
+
+- ``html-templates/``
+  	The html templates (using Jinja2 template engine) to be populated by processors.
+
+- ``processor-info.json``
+	Every processor is listed in this file, and will at least contain a regex pattern to match it's corresponding tag.
+	Most will also define required and optional parameters, these correspond to arguments in the tag's template.
+
+
+
+Creating a New Processor
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+To create a new processor, a good place to start is the `Extension API`_ page of the Python Markdown docs, or even the `source code`_ itself.
+
+.. _Extension API: https://pythonhosted.org/Markdown/extensions/api.html
+
+.. _source code: https://github.com/waylan/Python-Markdown
+
+
+There are several different kinds of processors in Python Markdown, each serving a slightly different purpose. We recommend reading the API docs to determine which processor best suits your purpose (Kordac currently makes use of ``preprocessor``, ``blockprocessor``, ``inlinepattern`` and ``treeprocessor``).
+
+The order of the processors matters and is defined when each processor is added to the ``OrderedDict`` in ``KordacExtension.py``.
+
+Each processor is independent of every other processor. If you have two processors in the pipeline that may overlap (e.g. codehilite and fencedcode), the second processor must handle whatever the first outputs, i.e. refrain from manipulating the output of the first processor to help the second.
+
+Every processor belongs in the ``processors/`` directory, but there are several other places they need to be listed, these are:
+
+- The frozenset of ``DEFAULT_PROCESSORS`` in ``Kordac.py``
+- The relevant list in ``extendMarkdown()`` in ``KordacExtension.py`` (see `OrderedDict in the Markdown API docs`_ for determining order)
+- The processor's template should be added to ``html-templates`` using the Jinja2 Template Engine syntax for variable parameters
+- The processor's relevant information (regex pattern, required parameters etc) should be included in ``processor-info.json``
+
+.. _OrderedDict in the Markdown API docs: https://pythonhosted.org/Markdown/extensions/api.html#ordereddict
+
+The new processors should also:
+
+- Be thoroughly tested (see the section below)
+- Have clear and accurate documentation. See the docs on other processors for the preferred format. Your docs should include:
+	- An example of the tag in markdown
+	- Required parameters
+	- Optional parameters
+	- Examples
+	- Examples of overriding the html
+
+We suggest writing the docs before you even write the processor itself as this will give you a clear idea of how a processor in Kordac should behave.
+
+
+Generally, every processor will have at least three methods (``__init__``, ``test`` and ``run``).
```

Comment:
Probably don't want to list this (or clarify it further) as it's covered in the Markdown extension documentation. 

Also `inlinepatterns` and `postprocessors` don't have `test` method.